pages/workspaces/[workspaceId]/repository-insights/[insightId]/activity.tsx

```diff
@@ -76,11 +77,27 @@ export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
   // Keeping this here so we are sure the page is not private before we fetch the social card.
   const ogImage = await fetchSocialCard(`insights/${insightId}`);
 
+  const { data: workspaceMembers } = await fetchApiData<any>({
+    path: `workspaces/${workspaceId}/members`,
+    bearerToken,
+    pathValidator: () => true,
+  });
+
+  const owners: string[] = Array.from(
+    workspaceMembers!.data,
```

Comment:
This is where the crash is happening when no workspace members are being returned. I'll fix this on the backend, but we probably want to also handle this on the client side just in case: if the `workspaceMembers` is empty, set the owners to an empty array.

```
- error pages/workspaces/[workspaceId]/repository-insights/[insightId]/dashboard.tsx (117:22) @ data
- error Error [TypeError]: Cannot read properties of null (reading 'data')
    at getServerSideProps$1 (webpack-internal:///./pages/workspaces/[workspaceId]/repository-insights/[insightId]/dashboard.tsx:160:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/jpmcb/workspace/opensauced/app/node_modules/@sentry/nextjs/cjs/common/utils/wrapperUtils.js:39:14
    at async /Users/jpmcb/workspace/opensauced/app/node_modules/@sentry/nextjs/cjs/common/utils/wrapperUtils.js:154:16
    at async Object.apply (/Users/jpmcb/workspace/opensauced/app/node_modules/@sentry/nextjs/cjs/common/wrapGetServerSidePropsWithSentry.js:44:33) {
  digest: undefined
}
  115 |
  116 |   const owners: string[] = Array.from(
> 117 |     workspaceMembers!.data,
      |                      ^
  118 |     (member: { role: string; member: Record<string, any> }) => {
  119 |       if (member.role === "owner") {
  120 |         return member.member.login;
```

---

@zeucapua you were right about the unauthorized:

```
[Nest] 53894  - 03/07/2024, 3:29:03 PM     LOG [OS] SELECT "workspaces"."id" AS "workspaces_id", "workspaces"."created_at" AS "workspaces_created_at", "workspaces"."updated_at" AS "workspaces_updated_at", "workspaces"."deleted_at" AS "workspaces_deleted_at", "workspaces"."name" AS "workspaces_name", "workspaces"."description" AS "workspaces_description", "workspaces"."is_public" AS "workspaces_is_public", "workspaces"."payee_user_id" AS "workspaces_payee_user_id", "workspace_members"."id" AS "workspace_members_id", "workspace_members"."user_id" AS "workspace_members_user_id", "workspace_members"."workspace_id" AS "workspace_members_workspace_id", "workspace_members"."created_at" AS "workspace_members_created_at", "workspace_members"."updated_at" AS "workspace_members_updated_at", "workspace_members"."deleted_at" AS "workspace_members_deleted_at", "workspace_members"."role" AS "workspace_members_role" FROM "workspaces" "workspaces" LEFT JOIN "workspace_members" "workspace_members" ON "workspace_members"."workspace_id"="workspaces"."id" AND ( "workspaces"."id"="workspace_members"."workspace_id" AND "workspace_members"."deleted_at" IS NULL) WHERE ( "workspaces"."id" = $1 ) AND ( "workspaces"."deleted_at" IS NULL ) -- Parameters: ["ff1160b5-cc32-45de-8e21-dd7b64f84a74"]
[Nest] 53894  - 03/07/2024, 3:29:03 PM    WARN [HTTP] GET /v2/workspaces/ff1160b5-cc32-45de-8e21-dd7b64f84a74/members 401 Unauthorized +147ms
```

this is a bug in the API and we should allow for public workspaces to serve the members.

Reply:
Unblocked here: https://github.com/open-sauced/api/pull/631